### PR TITLE
Updated file structure to support the Swift Package Manager

### PR DESCRIPTION
**Pull request by MichaelCampbell**
- Updated `Yoshi.Podspec` source_files and resources path
- Renamed top level Yoshi folder to Sources
- Added SPM link to README

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+//  Yoshi
+//
+//  Created by Michael Campbell on 4/15/16.
+//  Copyright © 2016 Prolific Interactive. All rights reserved.
+//
+
+import PackageDescription
+
+let package = Package(
+    name: "Yoshi"
```
Comment (DannyVancura) [inline Package.swift]: If I'm not mistaken, this should look more like this:
```swift
    name: "Yoshi",
    targets: [
        Target(
            name: "Yoshi"
        ),
        Target(
            name: "YoshiTests",
            dependencies: [.Target(name: "Yoshi")]
        )
    ]
```
This should create separate targets for the actual module and test targets.
The output currently: 
```shell
.../Yoshi.swift:9:8: error: no such module 'UIKit'
import UIKit
       ^
.../Yoshi.swift:9:8: error: no such module 'UIKit'
import UIKit
       ^
.../Yoshi.swift:9:8: error: no such module 'UIKit'
import UIKit
       ^
.../Yoshi.swift:9:8: error: no such module 'UIKit'
import UIKit
       ^
.../Yoshi.swift:9:8: error: no such module 'UIKit'
import UIKit
       ^
.../Yoshi.swift:9:8: error: no such module 'UIKit'
import UIKit
       ^
.../Yoshi.swift:9:8: error: no such module 'UIKit'
import UIKit
       ^
.../Yoshi.swift:9:8: error: no such module 'UIKit'
import UIKit
       ^
.../YoshiTests.swift:10:18: error: no such module 'Yoshi'
@testable import Yoshi
                 ^
<unknown>:0: error: build had 2 command failures
```

Where "Yoshi" seems to be referenced from within itself. When changing it as mentioned above, you get this:

```shell
.../Yoshi.swift:9:8: error: no such module 'UIKit'
import UIKit
       ^
.../Yoshi.swift:9:8: error: no such module 'UIKit'
import UIKit
       ^
.../Yoshi.swift:9:8: error: no such module 'UIKit'
import UIKit
       ^
.../Yoshi.swift:9:8: error: no such module 'UIKit'
import UIKit
       ^
.../Yoshi.swift:9:8: error: no such module 'UIKit'
import UIKit
       ^
.../Yoshi.swift:9:8: error: no such module 'UIKit'
import UIKit
       ^
.../Yoshi.swift:9:8: error: no such module 'UIKit'
import UIKit
       ^
.../Yoshi.swift:9:8: error: no such module 'UIKit'
import UIKit
       ^
<unknown>:0: error: build had 1 command failures
```

Where the "no such module errors" are caused by a [lack of support for iOS so far](https://github.com/apple/swift-package-manager/blob/master/Documentation/Package.swift.md#depending-on-apple-modules-eg-foundation).

**Comment by hkellaway** [comment]
@DannyVancura just curious - where did you get that output from ?

**Comment by DannyVancura** [comment]
I installed the latest development snapshot from [here](https://swift.org/builds/development/xcode/swift-DEVELOPMENT-SNAPSHOT-2016-04-12-a/swift-DEVELOPMENT-SNAPSHOT-2016-04-12-a-osx.pkg)
which comes with the Swift package manager and installed is as described [here](https://swift.org/getting-started/#installing-swift) (i.e. added the newly installed toolchain to the system path variable).
```shell
swift --version
Apple Swift version 3.0-dev (LLVM 752e1430fc, Clang 3987718dae, Swift 36739f7b57)
Target: x86_64-apple-macosx10.9

swift build --version
Apple Swift Package Manager 0.1
```

I then got the output by running `swift build` in the project folder (i.e. where `Package.swift` is located).

**Comment by Haud** [comment]
If Swift Package Manager currently has no support for iOS, why are we adding support for it?

**Comment by hkellaway** [comment]
if nothing else, it prepares us for when it does + improves visibility - it gives a  better ranking on CocoaPods and allows it to be listed in IBM Swift Package Catalog

**Comment by Haud** [comment]
So how do we test that it works?

**Comment by MichaelCampbell** [comment]
Closing this for now. Will revisit once we can fully test.

**Pull request closed without merging**